bot/actions/permissions.py

```diff
@@ -1,12 +1,17 @@
 from telegram.ext._contexttypes import ContextTypes
+from telegram import Update
 from bot.replies import replies
 from common.enums import Restriction
 from database import mongo
 from database.dbutils import dbutils
+from common import utils
+from typing import Optional
 
 
-async def restrict_to_admins(update, db_service):
-    chat_id = update.message.chat.id
+async def restrict_to_admins(update: Update, db_service: mongo.MongoService) -> None:
+    chat_id = utils.get_chat_id_from_update(update)
```

Comment:
Currently if `update.message` is None an error will be thrown in the logs, which will be useful for debugging. If we want to handle such errors properly we should log them. Same for similar changes in the other files. I suggest we create another PR for this refactoring as it's a bigger undertaking, and focus on adding types and cleaning up requirements.txt in this MR.